src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -140,6 +142,30 @@ Survey getOne() {
             }
             return surveyMapper.queryPage(DynamoSurvey.class, query).getResults();
         }
+        
+        private List<DynamoSurvey> queryBySurveyIdentifier() {
+            if (studyIdentifier == null) {
+                throw new IllegalStateException("Calculated the need to query by secondary index, but study identifier is not set");
+            }
+            DynamoSurvey hashKey = new DynamoSurvey();
+            hashKey.setStudyIdentifier(studyIdentifier);
+            hashKey.setIdentifier(surveyGuid.substring(11));
+            DynamoDBQueryExpression<DynamoSurvey> query = new DynamoDBQueryExpression<DynamoSurvey>();
+            query.withHashKeyValues(hashKey);
+            query.withConsistentRead(false);
+            query.withRangeKeyCondition("identifier", equalsString(surveyGuid.substring(11)));
+            if (published) {
+                query.withQueryFilterEntry(PUBLISHED_PROPERTY, equalsNumber("1"));
+            }
+            if (notDeleted) {
+                query.withQueryFilterEntry(DELETED_PROPERTY, equalsNumber("0"));
+            }
+            if (createdOn != 0L) {
+                query.withRangeKeyCondition(CREATED_ON_PROPERTY, equalsNumber(Long.toString(createdOn)));
+            }
+            
+            return surveyMapper.queryPage(DynamoSurvey.class, query).getResults();
```

Comment:
I think, for this to be effective, you have to use the index specified (in this case, the studyKey-identifier-index), via the Index object. Look at the use of DynamoIndexHelper in the DynamoScheduledActivityDao for an example of this.

Reply:
DynamoIndexHelper is buggy and should not be used. mapper.query() handles secondary indices just fine. See queryBySecondaryIndex().